azure-toolkit-libs/azure-toolkit-keyvault-lib/src/main/java/com/microsoft/azure/toolkit/lib/keyvault/secret/SecretVersionDraft.java

```diff
@@ -47,16 +57,54 @@ public SecretProperties createResourceInAzure() {
     @Override
     @AzureOperation(name = "azure/keyvault.update_secret_version.version", params = {"this.getName()"})
     public SecretProperties updateResourceInAzure(@Nonnull SecretProperties origin) {
-        final SecretAsyncClient secretClient = Objects.requireNonNull(getKeyVault().getSecretClient());
-        final Boolean isEnabled = ensureConfig().getEnabled();
+        final SecretDraft.Config config = ensureConfig();
+        final Boolean isEnabled = config.getEnabled();
         final boolean isModified = Objects.nonNull(isEnabled) && !Objects.equals(isEnabled, origin.isEnabled());
         if (isModified) {
-            origin.setEnabled(isEnabled);
-            return Objects.requireNonNull(secretClient.updateSecretProperties(origin).block(), "failed to update secret");
+            return updateSecretVersion(getKeyVault(), origin, config);
         }
         return origin;
     }
 
+    @Nonnull
+    public static SecretProperties createSecretVersion(@Nonnull final KeyVault keyVault, @Nonnull final SecretDraft.Config config) {
+        final SecretAsyncClient secretClient = keyVault.getSecretClient();
+        try {
+            final String value = config.getValue();
+            final KeyVaultSecret secret = secretClient.setSecret(config.getName(), value).block();
+            final SecretProperties properties = Objects.requireNonNull(secret).getProperties();
+            Optional.ofNullable(config.getEnabled()).ifPresent(properties::setEnabled);
+            Optional.ofNullable(config.getContentType()).ifPresent(properties::setContentType);
+            return Objects.requireNonNull(secretClient.updateSecretProperties(properties).block());
+        } catch (final Throwable t) {
+            // swallow all exceptions to prevent credential leakage
+            final Action<String> configure = getAccessPolicyConfiureAction(keyVault);
```

Comment:
Is it possible that the getAccessPolicyConfiureAction or getAccessPolicyLearnMoreAction throws?